### ISSUE TITLE
[all charts] add an automerge feature for stale approved charts

### ISSUE THREAD

**Issue by monotek** [prometheus-community/helm-charts#198]
**Is your feature request related to a problem? Please describe.**

There are several approved pull requests which are not merged, because repo admins are needed to do it.
Fore example: 

* https://github.com/prometheus-community/helm-charts/pull/114
* https://github.com/prometheus-community/helm-charts/pull/68

Imho these prs could be automerged if:

* it got an approve from one of the chart repo members
* it is older than x days (maybe 30?)

**Describe the solution you'd like**
A Github action which runs periodically and checks for such issues and automerges if acceptance criteria are met

If there is a "new chart" label (or something alike) it should alse be possible to merge without squash option to be able to get imported commits intio the repo.

@scottrigby @torstenwalter 
thoughts?

**Comment by scottrigby** [comment]
Hey @monotek 👋 

#68 requires an admin override, which we said we'd wait on until we have a policy. You're right that policy was merged [11 days ago](https://github.com/prometheus-community/helm-charts/pull/147/files#diff-5d0bfccf030216370e588b11377e76e1R14), so this one slipped through the cracks because it is a little weird. I'll go ahead and merge now. But to be clear, auto-merge wouldn't help here, since it's an exception requiring admin force-merge after not hearing from a maintainer for a period of time, right?

#114 went back and forth with change requests, and as of 2 days ago I made some requested changes myself to get it through. It's now failing CI, and until the reason is determined that shouldn't be merged. I'll try to re-run in case there was some temporary issue with CI, but I don't think an auto-merge would help this case either. Or do you see it differently?

**Comment by torstenwalter** [comment]
I am not sure if auto merge would solve your issue. Non of them is older than 39days now. So if you would wait for auto merge it would take even longer.

So far we just had two cases and the reasons are quite different.

One was a chart with only one maintainer and we said to wait for 2 weeks. Unfortunately after that we we missed to look after the PR again. There a reminder could help. From previous experience I would say that if it's important people will ping you again about the PR.

The other case was about a newly added chart and approval was mainly needed for CODEOWNERS file. I think the first time I looked at it either build was broken or there have been some other review comments, which blocked a merge.
For me it would be totally fine to create a separate PR just with the CODEOWNERS change. It would be easy to approve with a link to the PR, which contains the new chart. After that the maintainers could resolve remaining issues without approval from repo admins.

**Comment by monotek** [comment]
Ok, maybe we should wait how this evolves over time.

Naturally automation was my first thought :D

Feel free to close the ticket for now :)

**Comment by zanhsieh** [comment]
I think if auto advance chart version might be required before this.
